core/src/apps/cardano/sign_tx.py

```diff
@@ -125,117 +109,82 @@ async def sign_tx(ctx, msg, keychain: seed.Keychain):
 class Transaction:
     def __init__(
         self,
-        inputs: list,
-        outputs: list,
-        keychain,
+        inputs: List[CardanoTxInputType],
+        outputs: List[CardanoTxOutputType],
+        keychain: seed.Keychain,
         protocol_magic: int,
-        input_coins_sum: int,
-    ):
+        fee: int,
+        ttl: int,
+    ) -> None:
         self.inputs = inputs
         self.outputs = outputs
         self.keychain = keychain
-        # attributes have to be always empty in current Cardano
-        self.attributes = {}
+        self.fee = fee
+        self.ttl = ttl
 
         self.network_name = KNOWN_PROTOCOL_MAGICS.get(protocol_magic, "Unknown")
         self.protocol_magic = protocol_magic
-        self.input_coins_sum = input_coins_sum
 
-    def _process_outputs(self):
-        change_addresses = []
-        change_derivation_paths = []
-        output_addresses = []
-        outgoing_coins = []
-        change_coins = []
+    def _build_inputs(self) -> list:
```

Comment:
`_prepare_inputs_for_serialization` or just `_prepare_inputs` would imho give to the reader enough info so you don't need to name the return variable `inputs_for_cbor` which kind of exposes in the function a lower level concept that is not really needed to understand it

Reply:
How else can I name what I return? `prepared_inputs`? :shrug: It has no other use than to be used in CBOR, so I don't have a better name :(

Reply:
Imho `result` would do - the type would be clear from the type definition if updated and the most valuable information for the reader at that point is that this variable is the one meant to be returned, so no need to complicate it, but it's a matter of taste, of course